lib/ddtrace/contrib/sinatra/framework.rb

```diff
@@ -0,0 +1,78 @@
+module Datadog
+  module Contrib
+    # Instrument Sinatra.
+    module Sinatra
+      # Sinatra framework code, used to essentially:
+      # - handle configuration entries which are specific to Datadog tracing
+      # - instrument parts of the framework when needed
+      module Framework
+        # Configure Rack from Sinatra, but only if Rack has not been configured manually beforehand
+        def self.setup
+          Datadog.configure do |datadog_config|
+            sinatra_config = config_with_defaults(datadog_config)
+            activate_rack!(datadog_config, sinatra_config) unless Datadog.configuration.instrumented_integrations.key?(:rack)
+          end
+        end
+
+        def self.config_with_defaults(datadog_config)
+          datadog_config[:sinatra]
+        end
+
+        # Apply relevant configuration from Sinatra to Rack
+        def self.activate_rack!(datadog_config, sinatra_config)
+          datadog_config.use(
+            :rack,
+            service_name: sinatra_config[:service_name],
+            distributed_tracing: sinatra_config[:distributed_tracing],
```

Comment:
Also inspired by the Rails integration. TTBOMK this seems to be the only two common configuration keys.

Reply:
Given this is a fallback, it looks ok to me 👍.
The goal here is simply to make sure Rack visibility is present in tracers, but full configuration can be left to an explicit Rack configuration if the user so desires.